opencti-platform/opencti-front/src/components/FilterIconButtonContentWithRedirection.tsx

```diff
@@ -0,0 +1,57 @@
+import { Link } from 'react-router-dom';
+import React from 'react';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+import { resolveLink } from '../utils/Entity';
+import { truncate } from '../utils/String';
+import { useFormatter } from './i18n';
+import {
+  FilterIconButtonContentWithRedirectionStixObjectQuery,
+} from './__generated__/FilterIconButtonContentWithRedirectionStixObjectQuery.graphql';
+import { entityFilters } from '../utils/filters/filtersUtils';
+
+export const filterIconButtonContentWithRedirectionStixObjectQuery = graphql`
+    query FilterIconButtonContentWithRedirectionStixObjectQuery(
+        $id: String!
+    ) {
+        stixObjectOrStixRelationship(id: $id) {
+            ... on StixCoreObject {
+                id
+                parent_types
+                entity_type
+            }
+        }
+    }
+`;
+
+const FilterIconButtonContentWithRedirection = ({ filter, filterKey }: { filter: { id: string, value: string }, filterKey: string }) => {
+  const { t } = useFormatter();
+  const finalFilter = {
+    ...filter,
+    entity_type: undefined,
+  } as { id: string, value: string, entity_type?: string };
+  if (entityFilters.includes(filterKey)) {
+    const instanceData = useLazyLoadQuery<FilterIconButtonContentWithRedirectionStixObjectQuery>(filterIconButtonContentWithRedirectionStixObjectQuery, { id: filter.id });
+    finalFilter.entity_type = instanceData.stixObjectOrStixRelationship?.entity_type;
+  }
+  const link = finalFilter.entity_type ? resolveLink(finalFilter.entity_type) : null;
```

Comment:
The resolveLink method already handle the null case

Reply:
fixed